tests/unittests/MLTest.cpp

```diff
@@ -865,6 +865,89 @@ TEST_P(MLTest, convNetForImageRecognition) {
   }
 }
 
+/// Generate data for the regression test. Put a '1' in a random location in a
+/// clear tensor and report the coordinates of that pixel.
+static void generateRegressionTestData(Tensor &images, Tensor &labels) {
+  auto L = labels.getHandle<>();
+  auto image = images.getHandle<>();
+  unsigned numSamples = images.dims()[0];
+  image.clear(-0.5);
+
+  for (size_t i = 0; i < numSamples; i++) {
+    // Generate the X,Y coordinates to place our object.
+    size_t x = nextRandInt(0, 9);
+    size_t y = nextRandInt(0, 9);
+    L.at({i, 0}) = x;
+    L.at({i, 1}) = y;
+    image.at({i, x, y, 0u}) = 1;
+  }
+}
+
+/// This is the "Where's Waldo" test. We place a pixel in a tensor and the
+/// network reports the coordinate of the pixel.
+TEST_P(MLTest, testFindPixelRegression) {
+  const unsigned numSamples = 1000;
+  const unsigned batchSize = 10;
+
+  EE_.getConfig().learningRate = 0.01;
+  EE_.getConfig().batchSize = batchSize;
+  EE_.getConfig().momentum = 0.9;
+
+  auto &mod = EE_.getModule();
+  Function *F = mod.createFunction("main");
+
+  auto *input =
+      mod.createVariable(ElemKind::FloatTy, {batchSize, 10, 10, 1}, "input",
+                         VisibilityKind::Public, Variable::TrainKind::None);
+  auto *ex =
+      mod.createVariable(ElemKind::FloatTy, {batchSize, 2}, "coordinates",
+                         VisibilityKind::Public, Variable::TrainKind::None);
+
+  auto *FC0 = F->createFullyConnected("fc", input, 2);
```

Comment:
is that intentional that there are no hidden layers for the network? (i assume it'd learn better with more layers and we could make that 90% check tighter).